uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/validation/RoleValidator.java

```diff
@@ -0,0 +1,21 @@
+package org.uberfire.ext.security.management.api.validation;
+
+import org.jboss.errai.security.shared.api.Role;
+
+import javax.validation.ConstraintViolation;
+import java.util.Set;
+
+/**
+ * <p>The base validator class for a Role entity based on JSR303 Bean Validations.</p>
+ * <p>This provides validation logic for both backend and client sides, but you have to provide an instantiable class that provides the error message descriptions for each validation error supported.</p>
+ * 
+ * @since 0.8.0
+ */
+public abstract class RoleValidator implements EntityValidator<Role> {
+    
+    @Override
+    public Set<ConstraintViolation<Role>> validate(Role entity) {
+        return null;
```

Comment:
Throw an exception to indicate that it's not yet implemented instead of returning null and causes failures later?